core/unit_test/sycl/TestSYCL_InterOp_Streams.cpp

```diff
@@ -0,0 +1,123 @@
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Christian R. Trott (crtrott@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#include <sycl/TestSYCL_Category.hpp>
+#include <Test_InterOp_Streams.hpp>
+
+namespace Test {
+// Test Interoperability with SYCL Streams
+TEST(sycl, raw_sycl_queues) {
+  cl::sycl::default_selector device_selector;
+  cl::sycl::queue queue(device_selector);
+  Kokkos::InitArguments arguments{-1, -1, -1, false};
+  Kokkos::initialize(arguments);
+  int* p            = sycl::malloc_device<int>(100, queue);
+  using MemorySpace = typename TEST_EXECSPACE::memory_space;
+
+  {
+    TEST_EXECSPACE space0(queue);
+    Kokkos::View<int*, TEST_EXECSPACE> v(p, 100);
+    Kokkos::deep_copy(space0, v, 5);
+    int sum = 0;
+
+    Kokkos::parallel_for("Test::sycl::raw_sycl_queue::Range",
+                         Kokkos::RangePolicy<TEST_EXECSPACE>(space0, 0, 100),
+                         FunctorRange<MemorySpace>(v));
+    Kokkos::parallel_reduce("Test::sycl::raw_sycl_queue::RangeReduce",
+                            Kokkos::RangePolicy<TEST_EXECSPACE>(space0, 0, 100),
+                            FunctorRangeReduce<MemorySpace>(v), sum);
+    space0.fence();
+    ASSERT_EQ(6 * 100, sum);
+
+    Kokkos::parallel_for("Test::sycl::raw_sycl_queue::MDRange",
+                         Kokkos::MDRangePolicy<TEST_EXECSPACE, Kokkos::Rank<2>>(
+                             space0, {0, 0}, {10, 10}),
+                         FunctorMDRange<MemorySpace>(v));
+    space0.fence();
+    // FIXME_SYCL needs MDRangePolicy reduce
+#ifndef KOKKOS_ENABLE_SYCL
+    Kokkos::parallel_reduce(
+        "Test::sycl::raw_sycl_queue::MDRangeReduce",
+        Kokkos::MDRangePolicy<TEST_EXECSPACE, Kokkos::Rank<2>>(space0, {0, 0},
+                                                               {10, 10}),
+        FunctorMDRangeReduce<MemorySpace>(v), sum);
+    space0.fence();
+    ASSERT_EQ(7 * 100, sum);
+#endif
+
+    Kokkos::parallel_for("Test::sycl::raw_sycl_queue::Team",
+                         Kokkos::TeamPolicy<TEST_EXECSPACE>(space0, 10, 10),
+                         FunctorTeam<MemorySpace, TEST_EXECSPACE>(v));
+    space0.fence();
+    // FIXME_SYCL needs TeamPolicy reduce
+#ifndef KOKKOS_ENABLE_SYCL
+    Kokkos::parallel_reduce("Test::sycl::raw_sycl_queue::Team",
+                            Kokkos::TeamPolicy<TEST_EXECSPACE>(space0, 10, 10),
+                            FunctorTeamReduce<MemorySpace, TEST_EXECSPACE>(v),
+                            sum);
+    space0.fence();
+    ASSERT_EQ(8 * 100, sum);
+#endif
+  }
+  Kokkos::finalize();
+
+  // Try to use the queue after Kokkos' copy got out-of-scope.
```

Comment:
Explicitly mention `offset_streams` that his present in the HIP and CUDA counterpart of this unit test